### ISSUE TITLE
Annotation collections failing validation

### ISSUE THREAD

**Issue by sammyjava** [legumeinfo/datastore-issues#112]
--------------------------------------------------------------------------------
- [ ] Validating null collection Col.gnm9.ann10.WZQD
### INVALID: README.Col.gnm9.ann10.WZQD.yml is not present in collection Col.gnm9.ann10.WZQD
--------------------------------------------------------------------------------
- [ ] Validating cerca collection ISC453364.gnm1.ann1.HZJM
### INVALID: Required file type gene_models_main.gff3.gz is not present in ISC453364.gnm1.ann1.HZJM
--------------------------------------------------------------------------------
- [ ] Validating chafa collection MN87.gnm1.ann1.LWFM
### INVALID: chafa.MN87.gnm1.ann1.LWFM.gene_models_main.gff3.gz has an invalid ID attribute:
```
chafa.MN87.gnm1.scaffold10006_cov248       maker   CDS     -25618  -25584  0.0     1       Parent=chafa.MN87.gnm1.ann1.Chafa10006S05799
```
### INVALID: chafa.MN87.gnm1.ann1.LWFM.protein.faa.gz has an invalid sequence identifier in header:

`Chafa95793S30485`
--------------------------------------------------------------------------------
- [ ] Validating null collection Gy14.gnm1.ann1.Y8XX
### INVALID: README.Gy14.gnm1.ann1.Y8XX.yml is not present in collection Gy14.gnm1.ann1.Y8XX
--------------------------------------------------------------------------------
- [ ] Validating faial collection WAFC.gnm1.ann1.RTP9
### INVALID: faial.WAFC.gnm1.ann1.RTP9.gene_models_main.gff3.gz has an invalid ID attribute:
```
faial.WAFC.gnm1.scaffold1001_cov191        transdecoder    CDS     -4980   -4521   0.0     .       Parent=faial.WAFC.gnm1.ann1.Faial1001S00104
```
--------------------------------------------------------------------------------
- [ ] Validating glyma collection Lee.gnm1.ann1.6NZV
### INVALID: glyma.Lee.gnm1.ann1.6NZV.gene_models_main.gff3.gz record parent attribute is invalid; does the file need sorting?
```
glyma.Lee.gnm1.Gm01        phytozomev13    five_prime_UTR  37774   37783   0.0     .   ID=glyma.Lee.gnm1.ann1.GlymaLee.01G000100.1.five_prime_UTR.1;Parent=glyma.Lee.gnm1.ann1.GlymaLee.01G000100.1;pacid=41233089
```
--------------------------------------------------------------------------------
- [ ] Validating glyso collection F_IGA1003.gnm1.ann1.G61B
### INVALID: glyso.F_IGA1003.gnm1.ann1.G61B.gene_models_main.gff3.gz record parent attribute is invalid; does the file need sorting?
```
glyso.F_IGA1003.gnm1.chr11 maker   mRNA    6940    10361   0.0     .       ID=glyso.F_IGA1003.gnm1.ann1.SoyGsojaF_11G000200.1;Parent=glyso.F_IGA1003.gnm1.ann1.SoyGsojaF_11G000200;Name=SoyGsojaF_11G000200.1
```
--------------------------------------------------------------------------------
- [ ] Validating lotja collection MG20.gnm3.ann1.WF9B
### INVALID: lotja.MG20.gnm3.ann1.WF9B.gene_models_main.gff3.gz record ID attribute is missing or invalid:
```
lotja.MG20.gnm3.Lj4        protein_coding  gene    28879010        28884533        0.0     .       ID=gi|197209811|dbj|AB433810.1|.path1;Name=gi|197209811|dbj|AB433810.
1|.path1;Type="protein_coding"
```
### INVALID: lotja.MG20.gnm3.ann1.WF9B.protein.faa.gz has an invalid sequence identifier in header:
```AB433810.1```
### INVALID: lotja.MG20.gnm3.ann1.WF9B.protein_primary.faa.gz has an invalid sequence identifier in header:
```AB433810.1```
### INVALID: lotja.MG20.gnm3.ann1.WF9B.cds.fna.gz has an invalid sequence identifier in header:
```AB433810.1```
### INVALID: lotja.MG20.gnm3.ann1.WF9B.cds_primary.fna.gz has an invalid sequence identifier in header:
```AB433810.1```
### INVALID: lotja.MG20.gnm3.ann1.WF9B.mrna.fna.gz has an invalid sequence identifier in header:
```AB433810.1```
### INVALID: lotja.MG20.gnm3.ann1.WF9B.mrna_primary.fna.gz has an invalid sequence identifier in header:
```AB433810.1```
--------------------------------------------------------------------------------
- [ ] Validating medtr collection R108_HM340.gnm1.ann1.85YW
### INVALID: medtr.R108_HM340.gnm1.ann1.85YW.iprscan.gff3.gz has an invalid seqname:
```
medtr.R108_HM340.BZG31_000s000020.1        SUPERFAMILY     protein_match   78      413     0.0     .       Name=SSF51735;status=T;ID=match$2110_79_413;date=02-07-2021
```
--------------------------------------------------------------------------------
- [ ] Validating null collection Lovell.gnm2.ann1.S2ZZ
### INVALID: README.Lovell.gnm2.ann1.S2ZZ.yml is not present in collection Lovell.gnm2.ann1.S2ZZ
--------------------------------------------------------------------------------
- [ ] Validating null collection Heinz1706.gnm2_5.ann2_4.Q2DC
### INVALID: README.Heinz1706.gnm2_5.ann2_4.Q2DC.yml is not present in collection Heinz1706.gnm2_5.ann2_4.Q2DC
--------------------------------------------------------------------------------
- [ ] Validating tripr collection MilvusB.gnm2.ann1.DFgp
### INVALID: tripr.MilvusB.gnm2.ann1.DFgp.gene_models_main.gff3.gz record parent attribute is invalid; does the file need sorting?
```
tripr.MilvusB.gnm2.Tp1     ensembl five_prime_UTR  6545    6812    0.0     .       ID=tripr.MilvusB.gnm2.ann1.mRNA2509.utr1;Parent=tripr.MilvusB.gnm2.ann1.mRNA2509;Name=tripr.mRNA2509.utr1
```
--------------------------------------------------------------------------------
- [ ] Validating null collection PN40024.gnm12X.ann1.V31M
### INVALID: README.PN40024.gnm12X.ann1.V31M.yml is not present in collection PN40024.gnm12X.ann1.V31M

**Comment by adf-ncgr** [comment]
> Another check added after seeing that this file fouled up LegumeMine, which is just going to have to be the way it is because there is 0% chance of loading a LegumeMine without DS-sourced errors, because there are as many forms of errant data in the DS as there are galaxies in the new images from the James Webb. Whack-a-Mole is a trivial exercise in comparison.
> 
> ### Validating medtr collection A17.gnm5.ann1_6.L2RX
> * [ ]  medtr.A17.gnm5.ann1_6.L2RX.phytozome_10_2.HFNR.gfa.tsv.gz
> 
> ### INVALID: Gene family identifier 59253386 in medtr.A17.gnm5.ann1_6.L2RX.phytozome_10_2.HFNR.gfa.tsv.gz is not valid:
> ```
> medtr.A17.gnm5.ann1_6.MtrunA17Chr4g0042611 59253386        medtr.A17.gnm5.ann1_6.MtrunA17Chr4g0042611      1.6e-64 220.2   220.1
> ```

sorry about that, must have been generated before I baked the prefixes into the hmm db; should be fixed now, for whenever legumemine gets built again in future. BTW, I learned recently that "whack-a-mole" could be construed as "guacamole" in French; see https://justinehsmith.substack.com/p/notes-on-the-vibe-shift for more details if interested.

**Comment by adf-ncgr** [comment]
with respect to:
> INVALID: README.Col.gnm9.ann10.WZQD.yml is not present in collection Col.gnm9.ann10.WZQD
and several others of this ilk, these are from non-legume outgroup species included in the gene families. I'm not sure I remember why they got split out into Arabidopsis/thaliana/annotations style collections, but they seem to consist only of primary proteins and gfa files. It's possible I constructed the latter via post-gene family build assignment in order to deal with the fact that (as I understand it) when the families were built at most one representative gene from each outgroup species was included, meaning that oftentimes arabidopsis genes will seem to not be present; e.g. if I look in the family fastas in the gene families I find only 12757 arabidopsis records wheres the gfa file contains 24616 assignments. I suppose it would be worth loading these assignments into the mines (to enable queries based on non-legume genes with known functions). Presumably the explanation I've just reminded myself of can serve as the meat (or plant-based protein)  of their READMEs, when I get around to it.
